tools/clang/lib/SPIRV/CapabilityVisitor.cpp

```diff
@@ -63,10 +63,7 @@ void CapabilityVisitor::addCapabilityForType(const SpirvType *type,
     switch (floatType->getBitwidth()) {
     case 16: {
       // Usage of a 16-bit float type.
-      // It looks like the validator does not approve of Float16
-      // capability even though we do use the necessary extension.
-      // TODO: Re-enable adding Float16 capability below.
-      // addCapability(spv::Capability::Float16);
+      addCapability(spv::Capability::Float16);
       addExtension(Extension::AMD_gpu_shader_half_float, "16-bit float", loc);
```

Comment:
Do we still need `AMD_gpu_shader_half_float` ?

Reply:
You're right! We should not emit `AMD_gpu_shader_half_float` for any float16 usage. It should only be used when float16 is used in GLSL extended instruction set.

I'm working on another CL to fix this.